src/BaseControllerV2.ts

```diff
@@ -67,10 +69,10 @@ export class BaseController<S extends Record<string, any>> {
    *   object. Return a new state object or mutate the draft to update state.
    */
   protected update(callback: (state: Draft<S>) => void | S) {
-    const nextState = produce(this.internalState, callback) as S;
-    this.internalState = nextState;
+    const [nextState, patches] = produceWithPatches(this.internalState, callback);
+    this.internalState = nextState as S;
```

Comment:
It would be nice if this `as S` could be omitted, but when I leave it out I get an error I don't understand :thinking: